src/core/reaction_ensemble.cpp

```diff
@@ -1,1724 +1,1970 @@
-		//iff product_coefficients(i)-reactant_coefficients(i)<0, hide this number of reactant particles
-		if ( current_reaction->product_coefficients[i]-current_reaction->reactant_coefficients[i] >0) {
-			for(int j=0; j< current_reaction->product_coefficients[i]-current_reaction->reactant_coefficients[i] ;j++) {
-				int p_id=create_particle(current_reaction->product_types[i]);
-				p_ids_created_particles.push_back(p_id);
-			}
-		} else if (current_reaction->reactant_coefficients[i]-current_reaction->product_coefficients[i] >0) {
-			for(int j=0;j<current_reaction->reactant_coefficients[i]-current_reaction->product_coefficients[i];j++) {
-				append_particle_property_of_random_particle(current_reaction->reactant_types[i], hidden_particles_properties);			
-				hide_particle(hidden_particles_properties.back().p_id,current_reaction->reactant_types[i]);
-			}
-		}
-
-	}
-	//create or hide particles of types with noncorresponding replacement types
-	for(int i=std::min(current_reaction->len_product_types,current_reaction->len_reactant_types);i< std::max(current_reaction->len_product_types,current_reaction->len_reactant_types);i++ ) {
-		if(current_reaction->len_product_types<current_reaction->len_reactant_types){
-			//hide superfluous reactant_types particles
-			for(int j=0;j<current_reaction->reactant_coefficients[i];j++){
-				append_particle_property_of_random_particle(current_reaction->reactant_types[i], hidden_particles_properties);
-				hide_particle(hidden_particles_properties.back().p_id,current_reaction->reactant_types[i]);
-			}
-		} else {
-			//create additional product_types particles
-			for(int j=0;j<current_reaction->product_coefficients[i];j++){
-				int p_id= create_particle(current_reaction->product_types[i]);
-				p_ids_created_particles.push_back(p_id);
-			}
-		}
-	}	
-	
+void ReactionAlgorithm::make_reaction_attempt(
+    SingleReaction &current_reaction,
+    std::vector<StoredParticleProperty> &changed_particles_properties,
+    std::vector<int> &p_ids_created_particles,
+    std::vector<StoredParticleProperty> &hidden_particles_properties) {
+  // create or hide particles of types with corresponding types in reaction
+  for (int i = 0; i < std::min(current_reaction.len_product_types,
+                               current_reaction.len_reactant_types);
+       i++) {
+    // change std::min(reactant_coefficients(i),product_coefficients(i)) many
+    // particles of reactant_types(i) to product_types(i)
+    for (int j = 0; j < std::min(current_reaction.product_coefficients[i],
+                                 current_reaction.reactant_coefficients[i]);
+         j++) {
+      append_particle_property_of_random_particle(
+          current_reaction.reactant_types[i], changed_particles_properties);
+      replace_particle(changed_particles_properties.back().p_id,
+                       current_reaction.product_types[i]);
+    }
+    // create product_coefficients(i)-reactant_coefficients(i) many product
```

Comment:
i don't understand the comment

Reply:
the comment is valid and describes the logic of the replacement, deletion or creation of particles within the reaction ensemble algorithm.

Reply:
In doxygen comments you can actually use TeX ^^

Reply:
I am not sure how a comment like this would improve readability:
/**a(i)=product_coefficients(i)-reactant_coefficients(i)
*b(i)=product_coefficients(i)-reactant_coefficients(i)
*create a(i) many product particles if and only if a(i)>0,
* if and only if b(i)<0, hide b(i) many reactant particles */

Reply:
This for sure should go into a function with a descriptive name, actually maybe several functions...

Reply:
Feel free to do it